server/common/annotations.py

```diff
@@ -298,20 +337,31 @@ def write_labels(self, df, data_adaptor):
             dataset_id = uuid.uuid4()
             dataset = CellxGeneDataset(id=dataset_id, name=dataset_name)
             self.db.session.add(dataset)
+            self.db.session.commit()
 
-        uri = f"{self.directory_path}/{dataset_name}/{uid}/{timestamp}"
+        uri = f"{self.directory_path}/{dataset_name}/{self.user_id}/{timestamp}"
         if "s3" in uri:
```

Comment:
replace with "if uri.startswith("s3://"):"
this could be an error if the directory was something like "mydatasets3".

Reply:
👍 good call, thanks